### PR TITLE
Move documentation to GitHub 

### PR DESCRIPTION
**Pull request by mythsunwind**
Following the *Documentation-as-code* approach as described below i migrated the documentation from the wiki to this PR:
https://www.jenkins.io/blog/2019/10/21/plugin-docs-on-github/#how-to-enable-github-documentation-for-your-plugin

- [x] Convert documentation from Wiki to GitHub Markdown via Jenkins Wiki Exporter
- [x] Move images from wiki into repository & change markdown links for images
- [x] Change the <url> field in pom.xml so that it points to GitHub
- [x] Add badges to the page to improve look&feel and to provide quick links for users
- [x] Separate feature explanation from freestyle configuration
- [x] Add information for pipeline type configuration & update freestyle configuration

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,134 @@
...
+failed tests, aggregate and report flaky tests statistics and so on.
+
+**Note: Currently most features of this plugin are designed for Git,
+Maven and freestyle build projects.**
+
+## Install
+
+1. Get Jenkins.
+2. Install the Cucumber Reports plugin.
+3. Restart Jenkins.
```
Comment (vtintillier) [inline README.md:23]: Is this really necessary? Also this is about cucumber plugin 😃

File: README.md [hunk 2]
```diff
@@ -1,8 +1,134 @@
...
+
+A flaky test is a test which could fail or pass for the same
+configuration. Flaky tests could be harmful for developers because their
+failures do not always indicate bugs in the code.
+
+This plugin is designed to handle flaky tests, including re-running
+failed tests, aggregate and report flaky tests statistics and so on.
+
+**Note: Currently most features of this plugin are designed for Git,
+Maven and freestyle build projects.**
```
Comment (vtintillier) [inline README.md:17]: should we remove the _freestyle build_ part, now that you added support for pipeline? Or are there still limitations?

File: README.md [hunk 3]
```diff
@@ -1,8 +1,134 @@
...
+
+## Usage
+
+### Pipeline project type
+
+The plugin can be used in pipelines together with the [junit-realtime-test-reporter](https://www.jenkins.io/doc/pipeline/steps/junit-realtime-test-reporter/) plugin:
+
+    realtimeJUnit(keepLongStdio: true, testDataPublishers: [[$class: 'JUnitFlakyTestDataPublisher']], testResults: 'target/surefire-reports/TEST*.xml') {
+        sh 'mvn ...'
+    }
```
Comment (vtintillier) [inline README.md:90]: why are we mentioning that other plugin here? We can keep it, but we should first document "vanilla" usage with the standard junit reporter

**Review by vtintillier — COMMENTED** [review]
LGTM, minor comments

File: README.md (hunk 1 above)
Comment (mythsunwind) [reply README.md:23]: You caught me. I look at other READMEs for getting some ideas how to make ours better and copied it over from cucumber plugin

**Pull request merged**